inference-engine/tests/unit/inference_engine/saturated_cast_test.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+#include "common_test_utils/test_common.hpp"
+
+#include "saturated_cast.hpp"
+
+#include "ie_precision.hpp"
+
+using namespace InferenceEngine;
+
+class SaturatedCastTestsI64ToI32 : public CommonTestUtils::TestsCommon {
+public:
+    using fromType = typename PrecisionTrait<Precision::I64>::value_type;
+    using toType = typename PrecisionTrait<Precision::I32>::value_type;
+};
+
+TEST_F(SaturatedCastTestsI64ToI32, I64ToI32NonNarrowingPositive) {
+    EXPECT_EQ(saturated_cast<toType>(fromType{42}), toType{42});
```

Comment:
can we just include OpenCV header and compare with it for all types?

Reply:
Done.